### PR TITLE
Add periodic S3 export for DynamoDB tables

### PR DESCRIPTION
**Pull request by dennis-tra**
Hi @guseggert,

I had some time today and thought of giving the periodic S3 export a go. This should address issue #9.

Presumably, there are parts in this PR that are not in line with how things are done around here (e.g., resource naming, language/structure of lambdas, etc.). So, if that's the case, feel free to just take the parts you like and close this PR.

IIRC, the `main` table had a size of ~200GB. So, I guess, the lambda should remove the export `now() - 7d`. The point in time recovery still allows recovering data from the past ~30d.

### CODE REVIEW

**Review by guseggert — APPROVED** [review]
LGTM, I'll apply this once merged.

File: analytics.tf
Comment (guseggert) [inline analytics.tf]: we can add an expiration here so that all objects are automatically deleted after 7 days

Comment (dennis-tra) [reply analytics.tf]: Added a `aws_s3_bucket_lifecycle_configuration` resource to automatically expire objects after 7 days 👍

File: analytics.tf
Comment (guseggert) [inline analytics.tf]:
```suggestion
  bucket = "dynamodb-exports-${data.aws_caller_identity.current.account_id}-${data.aws_region.current.name}"
```

(consistent w/ the other bucket, also adding region+accountid in the bucket name prevents collisions, since S3 buckets across accounts+regions live in a global namespace)

**Comment by guseggert** [comment]
Do we want any sort of monitoring on this export process, so we know if it stops working? Or do you think we're fine without it?

**Comment by guseggert** [comment]
I've applied this with Terraform, and tested the Lambda to make sure it runs successfully, which it does now after fixing the bucket name. Just waiting to see if you want to monitor this, and then I'll merge.

**Comment by dennis-tra** [comment]
> [...] which it does now after fixing the bucket name.

Ah, thanks! 

---

It'd be good to monitor this. Is PL's Grafana instance wired up to CloudWatch of this AWS account? If so, I could build a tiny dashboard that shows lambda executions and also export events from DynamoDB (not sure if there're such events).

Did you think about another way of monitoring that?

**Comment by guseggert** [comment]
> Did you think about another way of monitoring that?

We have monitoring infrastructure but I'm not sure how to use it. Let me sync up with bifrost folks to find out.

**Comment by guseggert** [comment]
> Is PL's Grafana instance wired up to CloudWatch of this AWS account?

Yes it is, it is the "CloudWatch Hydras" data source in Grafana. Some of the Hydra graphs use it, see e.g. https://protocollabs.grafana.net/d/lNGQTv9Zz/hydra-boosters?orgId=1&editPanel=81

**Comment by guseggert** [comment]
So to summarize, we're not going to do automated alerting (overkill for now), and just stick with a dashboard that Dennis can look at. So nothing left to do here except merge.

**Pull request merged**
